crates/noirc_frontend/src/lexer/token.rs

```diff
@@ -401,8 +403,12 @@ impl AsRef<str> for Attribute {
     }
 }
 
+/// All keywords in Noir.
+/// Note that `self` is not present - it is a contextual keyword rather than a true one as it is
+/// only special within `impl`s. Otherwise `self` functions as a normal identifier.
+///
+/// To add a keyword, also make sure to add its string representation to Keyword::lookup_keyword.
```

Comment:
It would probably be useful to add a unit test similar to what we do for blackbox functions to check that there's a valid reverse mapping for any keywords.

See https://github.com/noir-lang/acvm/blob/f57ba57c2bb2597edf2b02fb1321c69cf11993ee/acir/src/circuit/black_box_functions.rs#L200-L211

Reply:
This test has been added in https://github.com/noir-lang/noir/pull/906